File: more-examples-pageobjects/java-testng/src/main/java/pageObjects/LoginPage.java
package pageObjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class LoginPage {

    WebDriver webDriver;

    @FindBy(name = "firstname")
    WebElement firstName;

    @FindBy(name = "lastname")
    WebElement lastName;

    @FindBy(name = "phone")
    WebElement phone;

    @FindBy(name = "email")
    WebElement email;

    @FindBy(name = "password")
    WebElement password;

    @FindBy(name = "confirmpassword")
    WebElement confirmPassword;

    @FindBy(css = "button.signupbtn.btn_full.btn.btn-primary.btn-block.btn-lg")
    WebElement signUp;

    public LoginPage(WebDriver webDriver) {
        this.webDriver = webDriver;
        PageFactory.initElements(this.webDriver, this);
    }

    public void visit() {
        this.webDriver.get("http://phptravels.net/register");
    }

    public void setFirstName(String firstName) {
        this.firstName.sendKeys(firstName);
    }

    public void setLastName(String lastName) {
        this.lastName.sendKeys(lastName);
    }

    public void setPhone(String phone) {
        this.phone.sendKeys(phone);
    }

    public void setEmail(String email) {
        this.email.sendKeys(email);
    }

    public void setPassword(String password) {
        this.password.sendKeys(password);
    }

    public void setConfirmPassword(String confirmPassword) {
        this.confirmPassword.sendKeys(confirmPassword);
    }

    public UserAccountPage clickSignUp() {
        this.signUp.click();
        return new UserAccountPage(this.webDriver);
    }

}

File: more-examples-pageobjects/js-mocha-chai/pageObjects/LoginPage.js
var webDriver = require('selenium-webdriver');
var By = webDriver.By;

var UserAccountPage = require('./UserAccountPage.js');

LoginPage = function LoginPage(driver) {
    this.driver = driver;
    this.firstName = By.name("firstname");
    this.lastName = By.name("lastname");
    this.phone = By.name("phone");
    this.email = By.name("email");
    this.password = By.name("password");
    this.confirmPassword = By.name("confirmpassword");
    this.signUp = By.css("button.signupbtn.btn_full.btn.btn-primary.btn-block.btn-lg");
};

LoginPage.prototype.visit = function() {
    this.driver.get('http://phptravels.net/register');
    return webDriver.promise.fulfilled(true);
};

LoginPage.prototype.setFirstName = function(fName) {
    this.driver.findElement(this.firstName).sendKeys(fName);
};

LoginPage.prototype.setLastName = function(lName) {
    this.driver.findElement(this.lastName).sendKeys(lName);
};

LoginPage.prototype.setPhone = function(phoneNumber) {
    this.driver.findElement(this.phone).sendKeys(phoneNumber);
};

LoginPage.prototype.setEmail = function(emailAddress) {
    this.driver.findElement(this.email).sendKeys(emailAddress);
};

LoginPage.prototype.setPassword = function(pwd) {
    this.driver.findElement(this.password).sendKeys(pwd);
};

LoginPage.prototype.setConfirmPassword = function(confirmPws) {
    this.driver.findElement(this.confirmPassword).sendKeys(confirmPws);
};

LoginPage.prototype.clickSignUp = function() {
    this.driver.findElement(this.signUp).click();
    return new UserAccountPage(this.driver);
};

module.exports = LoginPage;